app_dart/lib/src/model/appengine/branch.dart

```diff
@@ -0,0 +1,66 @@
+// Copyright 2021 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:cocoon_service/src/model/appengine/key_converter.dart';
+import 'package:gcloud/db.dart';
+import 'package:github/github.dart';
+import 'package:json_annotation/json_annotation.dart';
+
+part 'branch.g.dart';
+
+@Kind(name: 'Branch', idType: IdType.String)
+class Branch extends Model<String> {
+  Branch({Key<String>? key, required int lastActivity}) {
+    parentKey = key?.parent;
+    id = key?.id;
+  }
+
+  /// The timestamp (in milliseconds since the Epoch) of the last time
+  /// when current branch had activity.
+  @IntProperty(propertyName: 'lastActivity', required: true)
+  int? lastActivity;
+
+  /// [RepositorySlug] of where this commit exists.
+  RepositorySlug get slug => RepositorySlug.full(repository);
+
+  String get repository => key.id!.substring(0, key.id!.lastIndexOf('/'));
+
+  String get branch => key.id!.substring(key.id!.lastIndexOf('/') + 1);
+
+  @override
+  String toString() {
+    final StringBuffer buf = StringBuffer()
+      ..write('$runtimeType(')
+      ..write('id: $id')
+      ..write(', key: ${parentKey == null ? null : key.id}')
+      ..write(', branch: $branch')
+      ..write(', repository: $repository')
+      ..write(')');
+    return buf.toString();
+  }
+}
+
+/// The serialized representation of a [Branch].
+// TODO(tvolkert): Directly serialize [Branch] once frontends migrate to new serialization format.
+@JsonSerializable(createFactory: false, ignoreUnannotated: true)
```

Comment:
We should make this a proto instead, and create an adapter function for it.

Reply:
Yeah I spent quite a while trying to figure out how to replace the class with a proto and store it in db. Thanks for the extra info from Casey, I have now kept the class as it is, and dropped the protos under the frontend sub directory so that we can use the protos for frontend in the futures.